tensorpipe/test/channel/channel_test.h

```diff
@@ -19,6 +19,40 @@
 #include <tensorpipe/test/peer_group.h>
 #include <tensorpipe/transport/uv/context.h>
 
+class DataWrapper {
```

Comment:
Still wondering about this a bit. It's great that it allows us to run the CPU test suite on GPU channels, but how far can it go? Say, for now it happens to "hardcode" device 0 and the default stream. In some tests we'll certainly want to test with other devices and other streams. Those will certainly be GPU-only tests, so does that mean that we won't use this abstraction at all or is there a way to make it work?

Reply:
Sounds good